extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/destination/MSQDestination.java

```diff
@@ -30,5 +31,5 @@
 })
 public interface MSQDestination
 {
-  // No methods. Just a marker interface for deserialization.
+  ResultFormat getResultFormat();
```

Comment:
Lets document how this interface is used here.

Reply:
hmm this seems like the destination is marked with "x" result format which is not the case in taskReports as well as DurableStorageDestination. 

Since we have the context of the query passed to the task, is this change required. 
We can just look as the taskPayload to figure out the "resultFormat" passed on the post statement always . 

WDYT ?

Reply:
Changed